### PR TITLE
feat: add uniswap v3 trade helpers

### PR DESCRIPTION
**Pull request by KedziaPawel**
### Focus of this PR:
Add Uniswap v3 trade helpers to the SDK.

### Detailed summary:
- Added `IERC20` export in `packages/sdk/src/extensions/abis/index.ts`
- Updated snapshot values in `packages/sdk/src/utils/slippage.test.ts`, `packages/sdk/src/utils/rates.test.ts`, and `packages/sdk/src/utils/rates.test.ts`
- Removed a test in `packages/sdk/src/extensions/integrations/instances/aaveV3.test.ts`
- Updated imports in `packages/sdk/src/extensions/integrations/prepareUseIntegration/idleV4.test.ts`
- Added `UNISWAP_V3_ADAPTER` and `UNISWAP_V3_SWAP_ROUTER` constants in `packages/sdk/tests/constants.ts`
- Added `uniswapV3TakeOrderEncoding`, `UniswapV3TakeOrderArgs`, `encodeUniswapV3TakeOrderArgs`, and `decodeUniswapV3TakeOrderArgs` in `packages/sdk/src/extensions/integrations/instances/uniswapV3.ts`
- Updated types in `packages/sdk/src/extensions/integrations/integrationTypes.ts`
- Updated functions in `packages/sdk/src/extensions/integrations/prepareUseIntegration/prepareUseIntegration.ts`
- Added tests in `packages/sdk/src/extensions/integrations/instances/uniswapV3.test.ts` and `packages/sdk

> The following files were skipped due to too many changes: `packages/sdk/src/extensions/integrations/callOnIntegration/uniswapV3.test.ts`, `packages/sdk/src/constants/selectors.ts`, `packages/sdk/src/extensions/abis/IERC20.ts`, `packages/sdk/src/index.ts`, `packages/sdk/src/extensions/integrations/prepareUseIntegration/uniswapV3.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/sdk/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "build:cjs": "tsc --project tsconfig.build.json --module commonjs --verbatimModuleSyntax false --outDir ./dist/cjs && echo > ./dist/cjs/package.json '{\"type\":\"commonjs\"}'",
     "build:esm+types": "tsc --project tsconfig.build.json --module es2015 --outDir ./dist/esm --declaration --declarationMap --declarationDir ./dist/types && echo > ./dist/esm/package.json '{\"type\":\"module\", \"sideEffects\":false}'",
     "release": "pnpm build && ../../scripts/prerelease.sh ./package.json",
-    "test": "vitest",
+    "test": "vitest ",
```
Comment (fubhy) [inline packages/sdk/package.json:16]: Unintended change.

File: packages/sdk/src/index.ts [hunk 2]
```diff
@@ -606,6 +606,14 @@ export {
...
   type UniswapV2LiquidityRedeemArgs,
 } from "./extensions/integrations/instances/uniswapV2Liquidity.js";
+// ./extensions/integrations/instances/uniswapV2Liquidity.js
+
+export {
+  encodeUniswapV3TakeOrderArgs,
+  decodeUniswapV3TakeOrderArgs,
+  uniswapV3TakeOrderEncoding,
+  type UniswapV3TakeOrderArgs,
+} from "./extensions/integrations/instances/uniswapV3.js";
```
Comment (fubhy) [inline packages/sdk/src/index.ts:616]: Make sure you generate these exports with the script.

File: packages/sdk/tests/constants.ts [hunk 3]
```diff
@@ -11,10 +11,13 @@ export const COMPOUND_V2_ADAPTER = "0x6ce8095a692afF6698c3Aa8593BE3976B6b8743D"
 export const COMPOUND_V3_ADAPTER = "0xFaA9B9Cc98503F51A54F6038DfdD0E43AA0Ac98e" as const;
 export const IDLE_V4_ADAPTER = "0x474c8d4A0e53B7235C6f8fC27c9B6406a32Dd0b1" as const;
 export const UNISWAP_V2_LIQUIDITY_ADAPTER = "0xF78130AfedA6d9Df3394b34d36239aeC7FAe48d9" as const;
+export const UNISWAP_V3_ADAPTER = "0xeD6A08E05cB4260388dc7CC60Bc5fEFcCfab2793" as const;
 export const YEARN_VAULT_V2_ADAPTER = "0x7eA777F9F6ecBf4d03Dc5323d3F057B0730Fc34a" as const;
 export const MANAGEMENT_FEE = "0xfaf2c3db614e9d38fe05edc634848be7ff0542b9" as const;
 export const IDLE_V4_PRICE_FEED = "0x8Ddf1f3f0c13E099378B2B6F73cdB8F61526ed6F" as const;
 
+export const UNISWAP_V3_SWAP_ROUTER = "0xE592427A0AEce92De3Edee1F18E0157C05861564" as const;
```
Comment (fubhy) [inline packages/sdk/tests/constants.ts:19]: Let's organize and group these exports a bit.

**Review by fubhy — CHANGES_REQUESTED** [review]

File: packages/sdk/tests/constants.ts (hunk 3 above)
Comment (fubhy) [reply packages/sdk/tests/constants.ts:19]: Ok fair enough.

File: packages/sdk/src/index.ts (hunk 2 above)
Comment (KedziaPawel) [reply packages/sdk/src/index.ts:632]: done

File: packages/sdk/package.json (hunk 1 above)
Comment (KedziaPawel) [reply packages/sdk/package.json]: done

**Review by fubhy — APPROVED** [review]

**Pull request merged**